### PR TITLE
remove references to circle CI

### PR DESCRIPTION
**Pull request by brandonc**
## Output from tests
Including output from tests may require access to a TFE instance. Ignore this section if you have no environment to test against.

```
$ TFE_ADDRESS="https://example" TFE_TOKEN="example" TF_ACC="1" go test ./... -v -tags=integration -run TestFunctionsAffectedByChange

...
```

### CODE REVIEW

**Review by sebasslash — DISMISSED** [review]
:+1:

**Comment by brandonc** [comment]
@sebasslash I don't think its true anymore that people outside the org cannot run CI workflows. I'm going to reword some of this. They just have to be approved once.

File: README.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 Terraform Cloud/Enterprise Go Client
 ==============================
 
-[![Build Status](https://circleci.com/gh/hashicorp/go-tfe.svg?style=shield)](https://circleci.com/gh/hashicorp/go-tfe)
```
Comment (annawinkler) [inline README.md:4]: What do you think about adding a [Github Actions build badge](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge) instead?

**Review by annawinkler — COMMENTED** [review]
Maybe we could add a badge for GH actions? 💖

**Review by annawinkler — APPROVED** [review]
👍 My comment was addressed in another [PR](https://github.com/hashicorp/go-tfe/pull/509)

**Pull request merged**